src/main/java/eu/dissco/backend/service/MachineAnnotationServiceService.java

```diff
@@ -38,15 +37,13 @@ public class MachineAnnotationServiceService {
   private final ObjectMapper mapper;
 
   private boolean checkIfMasComplies(JsonNode jsonNode,
-      MachineAnnotationServiceRecord masRecord) {
-    var filters = masRecord.mas().targetDigitalObjectFilters();
-    var fields = filters.fields();
+      MachineAnnotationService machineAnnotationService) {
+    var filters = machineAnnotationService.getOdsTargetDigitalObjectFilter();
+    var fields = filters.getAdditionalProperties();
     var complies = true;
-    while (fields.hasNext() && complies) {
-      var field = fields.next();
-      var fieldKey = field.getKey();
-      var allowedValues = mapper.convertValue(field.getValue(), new TypeReference<List<Object>>() {
-      });
+    for (var stringObjectEntry : fields.entrySet()) {
```

Comment:
using the for instead of while loop, you don't break immediately if complies=false. add break statement?